scripts/dts/gen_defines.py

```diff
@@ -440,7 +440,14 @@ def irq_name_alias(irq, cell_name):
         # We ignore the controller for now
         for cell_name, cell_value in irq.specifier.items():
             ident = "IRQ_{}".format(irq_i)
-            if cell_name != "irq":
+            if cell_name == "irq":
+                # TODO: ADD COMMENT ABOUT ZEPHYR IRQ NUM
```

Comment:
Yeah, some more comments would be nice here, to explain how `irq` is special, why the values are concatenated, etc. Should document the special `irq` behavior in `binding-template.yaml` as well (or in the `.rst`).

Does the code make any assumptions, like all interrupts being sent to the same place? Would be nice to mention those as well.

Reply:
Not sure what to document in `binding-template.yaml`, before this change we should document how multi-level IRQs are "hacked" in.  This fixes it so its more natural and reflects the hardware in the DTS (and remove's the Zephyr specific hack/encoding).

Reply:
Maybe it could be a link to that interrupts page on https://docs.zephyrproject.org/latest/guides/dts/index.html instead, if it's documented there.

Doesn't have to go in this PR though.

Reply:
that make's sense, to have a section there about IRQ define generation.  Opened issue #18137 for this.